pkg/controllers/disruption/consolidation.go

```diff
@@ -101,61 +97,129 @@ func (c *consolidation) ShouldDisrupt(_ context.Context, cn *Candidate) bool {
 		c.recorder.Publish(disruptionevents.Unconsolidatable(cn.Node, cn.NodeClaim, fmt.Sprintf("NodePool %q has consolidation disabled", cn.nodePool.Name))...)
 		return false
 	}
-	return true
-}
-
-// sortCandidates sorts candidates by disruption cost (where the lowest disruption cost is first) and returns the result
-func (c *consolidation) sortCandidates(candidates []*Candidate) []*Candidate {
-	sort.Slice(candidates, func(i int, j int) bool {
-		return candidates[i].disruptionCost < candidates[j].disruptionCost
-	})
-	return candidates
+	return cn.NodeClaim.StatusConditions().GetCondition(v1beta1.Underutilized).IsTrue()
 }
 
-// computeConsolidation computes a consolidation action to take
+// ComputeCommand computes a consolidation action to take
 //
 // nolint:gocyclo
-func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...*Candidate) (Command, pscheduling.Results, error) {
-	var err error
-	// Run scheduling simulation to compute consolidation option
-	results, err := SimulateScheduling(ctx, c.kubeClient, c.cluster, c.provisioner, candidates...)
-	if err != nil {
-		// if a candidate node is now deleting, just retry
-		if errors.Is(err, errCandidateDeleting) {
-			return Command{}, pscheduling.Results{}, nil
+func (c *Consolidation) ComputeCommand(ctx context.Context, disruptionBudgetMapping map[string]int, candidates ...*Candidate) (Command, pscheduling.Results, error) {
```

Comment:
Do we only validate that a command respects disruption budgets when computing it in the underutilization controller? Isn't it likely that the cluster state has changed enough that we would need to revalidate by the time the command is actually ready for execution?